splice_spark/src/main/spark2.1/com/splicemachine/spark/splicemachine/SplicemachineContext.scala

```diff
@@ -227,17 +215,37 @@ class SplicemachineContext(url: String) extends Serializable {
     internalConnection.createStatement().executeUpdate(sqlText)
   }
 
+  def upsert(dataFrame: DataFrame, schemaTableName: String): Unit = {
```

Comment:
woohoo

Reply:
This should have been separated but I am showing my age...